src/store/ethereum/wallets/walletlink.ts

```diff
@@ -0,0 +1,60 @@
+import { ethers } from "ethers"
+import Web3 from "web3"
+import WalletLink from "walletlink"
+
+import { WalletType, EthereumConfig, IWalletProvider } from "../types"
+
+export const WalletLinkAdapter: WalletType = {
+    id: "walletlink",
+    name: "WalletLink",
+    detectable: true,
+    isMultiAccount: false,
+    desktop: true,
+    mobile: true,
+    detect() {
+        return true
+    },
+    async createProvider(config: EthereumConfig): Promise<IWalletProvider> {
+        const chainId = parseInt(config.networkId, 10)
+        // FIXME: Hardcoded to mainnet right now since WalletLink API is half-baked and doesn't provide
+        // a way for the user to specify the network within the wallet app.
+        if (chainId !== 1) {
+            alert("WalletLink only works with mainnet for the moment " + config.networkId)
```

Comment:
It does let me pick a different network in the Coinbase wallet app... you sure this doesn't work with other chain IDs?

Reply:
Removed